packages/core/src/utils/classy/classy.ts

```diff
@@ -19,73 +18,39 @@ type Template = [TemplateStringsArray, ...ClassNames[]];
  * // 'foo-1g4k53 global-1 global-2 className disabled'
  */
 export const classy = (...classNames: ClassNames[]) =>
-  classNames
-    .flat(10)
-    .flatMap((className) =>
-      className instanceof Object ? getTruthyKeys(className) : className
-    )
-    .filter(Boolean)
-    .join(' ');
+  classNames.flat(10).flatMap(truthyNames).join(' ');
 
 /**
  * Prefixes component names with a predetermined identifier.
  *
- * Can be used as tagged template literal, separated by spaces.
- *
  * @param components Names to prefix.
  *
  * @example
- * c('button') || c`button`;
+ * c('button');
  * // ['ods-button']
  */
-export const c = mapWithPrefix('ods-');
+export const c = withPrefix('ods-');
 
 /**
  * Prefixes modifier names with a predetermined identifier.
  *
- * Can be used as tagged template literal, separated by spaces.
- *
  * @param modifiers Names to prefix.
  *
  * @example
- * m('action', 'primary') || m`action primary`;
+ * m('action', 'primary');
  * // ['-action', '-primary']
  */
-export const m = mapWithPrefix('-');
-
-const fromTemplate = (array: Template | ClassNames[]) =>
-  isTemplate(array)
-    ? array[0]
-        .map((value, i) => value + classy(array[i + 1] as ClassNames))
-        .join('')
-        .split(' ')
-    : array;
-
-const getTruthyKeys = (obj: Record<string, unknown>): string[] =>
-  Object.entries(obj)
-    .filter(([, value]) => !!value)
-    .map(([key]) => key);
-
-const isTemplate = (template: Template | unknown[]): template is Template =>
-  Array.isArray(template[0]) &&
-  Array.isArray(((template[0] as unknown) as TemplateStringsArray).raw);
-
-const mapKey = <V>(
-  obj: Record<string, V>,
-  mapFn: (key: string) => string
-): Record<string, V> =>
-  Object.fromEntries(
-    Object.entries(obj).map(([key, value]) => [mapFn(key), value])
-  );
-
-function mapWithPrefix(prefix: string) {
-  return (...array: ClassNames[] | Template) =>
-    fromTemplate(array)
-      .flat(10)
-      .filter(Boolean)
-      .flatMap((className) =>
-        className instanceof Object
-          ? getTruthyKeys(mapKey(className, (key) => `${prefix}${key}`))
-          : `${prefix}${className}`
-      );
+export const m = withPrefix('-');
+
+const truthyNames = (value: ClassName | ClassNameRecord) =>
+  (value instanceof Object
+    ? Object.entries(value)
+        .filter(([, value]) => !!value)
+        .map(([key]) => key)
+    : [value]
+  ).filter(Boolean);
+
+function withPrefix(prefix: string) {
```

Comment:
Curious, why did you choose `truthyNames` to be an arrow function, and this a regular one?

Reply:
I prefer arrow functions in general, except when the function has a body (multiple expressions).

However this one didn't leave me much choice as it's used in lines 32 and 43 so it needs to be hoisted.

Reply:
I guess something to think about - maybe we would want to standartise this, e.g. use one kind only.

Reply:
If we were to, I suppose they'd all need to be regular functions, because sometimes we need the hoisting.

We can have guidelines for sure, extending eslint's rules.
There are many instances where arrow functions are much terser.